homeassistant/components/srp_energy/config_flow.py

```diff
@@ -40,46 +40,53 @@ class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
 
     VERSION = 1
 
-    async def async_step_user(
-        self, user_input: dict[str, Any] | None = None
-    ) -> FlowResult:
-        """Handle a flow initialized by the user."""
-        errors = {}
-        default_title: str = DEFAULT_NAME
-
-        if self._async_current_entries():
-            return self.async_abort(reason="single_instance_allowed")
-
-        if self.hass.config.location_name:
-            default_title = self.hass.config.location_name
-
-        if user_input:
-            try:
-                await validate_input(self.hass, user_input)
-            except ValueError:
-                # Thrown when the account id is malformed
-                errors["base"] = "invalid_account"
-            except InvalidAuth:
-                errors["base"] = "invalid_auth"
-            except Exception:  # pylint: disable=broad-except
-                LOGGER.exception("Unexpected exception")
-                return self.async_abort(reason="unknown")
-            else:
-                return self.async_create_entry(title=default_title, data=user_input)
-
+    @callback
+    def _show_form(self, errors: dict[str, Any]) -> FlowResult:
+        """Show the form to the user."""
+        LOGGER.debug("Show Form")
         return self.async_show_form(
             step_id="user",
             data_schema=vol.Schema(
                 {
+                    vol.Required(
+                        CONF_NAME, default=self.hass.config.location_name
```

Comment:
Don't ask for the name here. Just default to the location name. The user can change the config entry title afterwards if they want.